apex-squid/src/main/java/org/fundacionjala/enforce/sonarqube/apex/api/ApexGrammar.java

```diff
@@ -267,39 +346,89 @@ private static void typeClass(ApexGrammarBuilder grammarBuilder) {
      *
      * @param grammarBuilder ApexGrammarBuilder parameter.
      */
-    private static void classOrInterfaceBodyDeclaration(ApexGrammarBuilder grammarBuilder) {
-        grammarBuilder.rule(CLASS_OR_INTERFACE_BODY_DECLARATION).is(
-                MODIFIERS,
-                METHOD_DECLARATION
+    private static void fieldDeclaration(ApexGrammarBuilder grammarBuilder) {
+        grammarBuilder.rule(FIELD_DECLARATION).is(
+                grammarBuilder.firstOf(
+                        METHOD_DECLARATION,
+                        CONSTRUCTOR_DECLARATION,
+                        VARIABLE_DECLARATION)
+        );
+    }
+
+    /**
+     * It is responsible for setting the rules for parameters.
+     *
+     * @param grammarBuilder ApexGrammarBuilder parameter.
+     */
+    private static void parameter(ApexGrammarBuilder grammarBuilder) {
+        grammarBuilder.rule(PARAMETER).is(
+                TYPE,
+                IDENTIFIER,
+                grammarBuilder.optional(BRACKETS)
+        );
+
+    }
+
+    /**
+     * It is responsible for setting the rules for the optional argument list.
+     *
+     * @param grammarBuilder ApexGrammarBuilder parameter.
+     */
+    private static void parameterListOptional(ApexGrammarBuilder grammarBuilder) {
+        grammarBuilder.rule(PARAMETER_LIST_OPTIONAL).is(COMMA, PARAMETER);
+    }
+
+    /**
+     * It is responsible for setting the rules for the argument list.
+     *
+     * @param grammarBuilder ApexGrammarBuilder parameter.
+     */
+    private static void parameterList(ApexGrammarBuilder grammarBuilder) {
+        grammarBuilder.rule(PARAMETER_LIST).is(
+                PARAMETER,
+                grammarBuilder.optional(PARAMETER_LIST_OPTIONAL)
         );
     }
 
     /**
+     * It is responsible for managing the method name.
+     *
+     * @param grammarBuilder ApexGrammarBuilder parameter.
+     */
+    private static void methodName(ApexGrammarBuilder grammarBuilder) {
+        grammarBuilder.rule(METHOD_NAME).is(IDENTIFIER);
+    }
+
+    /**
      * Creates rules to the last line of the method and the completion of the method.
      *
      * @param grammarBuilder ApexGrammarBuilder parameter.
      */
     private static void methodDeclaration(ApexGrammarBuilder grammarBuilder) {
         grammarBuilder.rule(METHOD_DECLARATION).is(
-                RESULT_TYPE,
-                RBRACE
+                MODIFIER,
+                TYPE,
+                METHOD_NAME,
+                LPAREN,
+                grammarBuilder.optional(PARAMETER_LIST),
+                RPAREN,
+                STATEMENT_BLOCK
         );
     }
 
     /**
-     * Creates rules for the header of a method.
+     * Creates rules to the last line of the method and the completion of the method.
```

Comment:
Creates rules to the last line of the method and the completion of the method.

Creates rules for method's last line and completion